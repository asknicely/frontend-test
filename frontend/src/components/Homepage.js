import React, { useState, useEffect } from 'react'
import axios from 'axios'
import Loader from './Loader'

const Homepage = (props) => {

	const [data, setData] = useState({
		content: '',
		isLoaded: false
	})

	const fetchData = async () => {
		const url = `${process.env.REACT_APP_BACKEND_URL}`
		const response = await axios.get(url)
		if (response.data.status === 200) setData({
			content: response.data.contents,
			isLoaded: true
		})
	}

	useEffect(() => {
		document.title = `Todos App Homepage`
		if (!data.isLoaded) fetchData()
	}, [data])


	if (!data.isLoaded) return (<Loader />)
	else {
		return (
			<React.Fragment>
				<h1>README</h1>
				<pre className="well">{data.content}</pre>
			</React.Fragment>
		)
	}

}

export default Homepage