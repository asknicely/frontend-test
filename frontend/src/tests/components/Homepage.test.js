import React, { useReducer } from 'react';
import { shallow, mount, render } from 'enzyme';
import Homepage from '../../components/Homepage';

import mockAxios from "axios";
jest.mock('axios');

test('should render Homepage correctly', () => {

	const mockedResponse = {
		status: 200,
		statusTxt: 'ok',
		contents: 'This is a mocked axios response for unit testing on Homepage component'
	};

	mockAxios.get.mockImplementationOnce(() =>
		Promise.resolve(mockedResponse)
	);	

	// axios.get.mockResolvedValue(mockedResponse);

	const wrapper = shallow(<Homepage />);
	expect(wrapper).toMatchSnapshot();
	
});
