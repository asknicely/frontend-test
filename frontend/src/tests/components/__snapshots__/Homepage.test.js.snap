// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Homepage correctly 1`] = `<Loader />`;
